ocaml/xapi/xapi_vbd.ml

```diff
@@ -132,8 +132,16 @@ let unplug_force_no_safety_check = unplug_force
 let autodetect_mutex = Mutex.create ()
 
 (** VBD.create doesn't require any interaction with xen *)
-let create  ~__context ~vM ~vDI ~userdevice ~bootable ~mode ~_type ~unpluggable ~empty
-    ~other_config ~qos_algorithm_type ~qos_algorithm_params =
+let create  ~__context ~vM ~vDI ~device ~userdevice ~bootable ~mode ~_type ~unpluggable ~empty
+    ~other_config ~currently_attached ~qos_algorithm_type ~qos_algorithm_params =
+
+  if device <> "" || currently_attached then begin
+    match (Db.VM.get_power_state ~__context ~self:vM) with
+      | `Suspended -> ()
+      | _ -> raise (Api_errors.(Server_error (
+          vm_bad_power_state, ["Plugged VBD creation only allowed for suspended VM"])
```

Comment:
You should also say which VM the error is for. This is not yet enforced by the type checker, but see datamodel_errors.ml you need 3 arguments here, not 1: `error Api_errors.vm_bad_power_state ["vm"; "expected"; "actual"]`
See examples elsewhere in the code:
`Api_errors.vm_bad_power_state [ Ref.string_of vm; expected; actual `

Reply:
Done! :)